pkg/scheduler/objects/preemption.go

```diff
@@ -749,6 +749,64 @@ func (qps *QueuePreemptionSnapshot) GetRemainingGuaranteed() *resources.Resource
 	return resources.ComponentWiseMin(remaining, parentResult)
 }
 
+func (qps *QueuePreemptionSnapshot) GetPreemptableResource() *resources.Resource {
+	if qps == nil {
+		return nil
+	}
+	parentPreemptableResource := qps.Parent.GetPreemptableResource()
+	used := qps.AllocatedResource.Clone()
+
+	// No usage, so nothing to preempt
+	if used.IsEmpty() {
+		return nil
+	}
+	used.SubOnlyExisting(qps.PreemptingResource)
+
+	// Calculate preemptable resource. +ve means Over utilized, -ve means Under utilized, 0 means correct utilization
+	guaranteed := qps.GuaranteedResource.Clone()
+	actualPreemptableResource := used
```

Comment:
not needed, can be used directly
maybe rename `used` to `actual` and drop `actualPreemptableResource`